Update Boost to 1.65 and make it default

abbradar opened the pull request:
###### Motivation for this change

Update the default package. It brings fixes for NVCC so that we can use new CUDA toolkit 9.0 with default Boost (which we need due to GCC 6 default). Also NumPy support is improved.

I've made packages that overrode Boost to a newer one use the default version. RStudio update is needed to fix build.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I work on a system built with this Boost for some time, but don't have resources to test all packages.

abbradar committed:
boost165: init at 1.65.1

abbradar committed:
boost: make 1.65 the default

abbradar committed:
boost: add numpy support

Not built by default.

Also add enablePython flag (enabled by default) and cleanup a bit.

abbradar committed:
rstudio: enable parallel building

Also cleanup inputs.

peterhoeg commented on pkgs/applications/editors/rstudio/default.nix:
```diff
@@ -1,20 +1,20 @@
   name = "RStudio-${version}";
 
-  buildInputs = [ cmake boost163 zlib openssl R qt5.full qt5.qtwebkit qt5.qtwebchannel libuuid unzip ant jdk makeWrapper pandoc ];
+  buildInputs = [ cmake boost zlib openssl R qt5.full qt5.qtwebkit qt5.qtwebchannel libuuid unzip ant jdk makeWrapper pandoc ];
```
You want to clean up the nativeBuildInputs while at it?

abbradar replied:
Good idea, done.

abbradar commented:
Ugh, Github requested reviews automatically because I accidentially pushed the wrong branch. Sorry.

peti commented on pkgs/applications/altcoins/default.nix:
```diff
@@ -1,4 +1,4 @@
-{ callPackage, boost155, boost162, boost163, openssl_1_1_0, haskellPackages, darwin, libsForQt5 }:
+{ callPackage, boost155, boost162, openssl_1_1_0, haskellPackages, darwin, libsForQt5 }:
```
This build needs both boost155 and boost162?

abbradar replied:
It's a subpackage tree, some altcoins seem to need older Boosts.

peti reviewed (approved):
I like it. There's certainly going to be some fallout because of this change,so when the PR is merged there'll be another round of committing fixes to apps that break because of the new version. :-/

abbradar commented:
Let's merge this in several days if noone objects; I'll be watching Hydra for possible fallout.

abbradar committed:
steam-run-native: init

Refactor Steam so that native steam-run can be built as a free package and
expose it so that Hydra builds needed i686 packages.

Lassulus committed:
encrypted devices: don't override default cryptoModules

Lassulus committed:
encrypted devices: provide working example

bjornfor committed:
nixos/tests: unbreak prometheus test

Commit 271d3f7a432f ("prometheus service: globalConfig.labels is obsolete")
removed globalConfig.labels. Update the test config accordingly.

bjornfor committed:
nixos/tests: enable prometheus test

As far as I can tell, we've had this test since 2015, but it was never
enabled.

Mic92 committed:
Merge pull request #30477 from Lassulus/encrypted-devices

Encrypted devices: don't override crytpoModules, fix example

thoughtpolice committed:
aiger: fix stupid incorrect download link

Signed-off-by: Austin Seipp <aseipp@pobox.com>

abbradar committed:
buildFHSEnv: fix NIX_* compiler flags

This is needed now after #27672.

Commit:
sc2-headless: init at 3.17

This packages up the Starcraft II headless Linux client provided by
Blizzard for machine learning research.

Ericson2314 committed:
Merge pull request #30481 from obsidiansystems/sc2-headless

sc2-headless: init at 3.17

Ericson2314 committed:
sc2-headless: License is unfree

I wrongly assumed that was the default when reviewing the PR.

abbradar committed:
Merge branch 'master' into HEAD
